src/python/pants/util/dirutil.py

```diff
@@ -480,19 +528,17 @@ def touch(path, times=None):
     :times Either a tuple of (atime, mtime) or else a single time to use for both.  If not
            specified both atime and mtime are updated to the current time.
   """
-  if times:
-    if len(times) > 2:
-      raise ValueError('times must either be a tuple of (atime, mtime) or else a single time value '
-                       'to use for both.')
-
-    if len(times) == 1:
-      times = (times, times)
-
+  if isinstance(times, tuple) and len(times) > 2:
+    raise ValueError(
+      "`times` must either be a tuple of (atime, mtime) or else a single time to use for both."
+    )
+  if isinstance(times, int):
+    times = (times, times)
```

Comment:
I believe the original implementation was implemented incorrectly. If the user passed just one value for `times`, it would still have to be wrapped in an iterable, like: `touch('test.txt', [100])`. This would fail: `touch('test.txt', 100)`. Per the comment, this does not seem intended so I made the change.